app/src/viewmodels/camera/CameraViewModel.tsx

```diff
@@ -20,33 +25,34 @@ export default function useCameraViewModel(
   navigation: any,
   route: any,
 ) {
-  const [facing, setFacing] = useState<CameraType>("back");
-  const [permission, requestPermission] = useCameraPermissions();
+  // Camera state
   const camera = useRef<CameraView>(null);
-  const [picture, setPicture] = useState<CameraCapturedPicture>();
+  const cameraPictureOptions: CameraPictureOptions = { base64: true };
+  const [permission, requestPermission] = useCameraPermissions();
   const [isCameraEnabled, setIsCameraEnabled] = useState(true);
-  const [flashMode, setFlashMode] = useState<FlashMode>("off");
+  const [picture, setPicture] = useState<CameraCapturedPicture>();
+
+  const [facing, setFacing] = useState<CameraType>("back");
+
   const [isFlashEnabled, setIsFlashEnabled] = useState(false);
-  const [zoom, setZoom] = useState(0);
-  const [lastZoom] = useState(0);
 
-  const group_id = route.params?.group_id;
+  // Location state
+  const [location, setLocation] = useState<LocationObject | null>(null);
+  const [isLocationEnabled, setIsLocationEnabled] = useState(true);
 
-  const cameraPictureOptions: CameraPictureOptions = { base64: true };
+  // Challenge state
+  const [description, setDescription] = useState("");
 
-  // Calculate the zoom level
-  const calculateZoom = (event: any, velocity: number, outFactor: number) => {
-    const scaleFactor = Platform.OS === "ios" ? 0.01 : 25;
-    const reduceFactor = Platform.OS === "ios" ? 0.02 : 50;
+  const group_id = route.params?.group_id;
 
-    if (velocity > 0) {
-      return zoom + event.scale * velocity * scaleFactor;
-    } else {
-      return (
-        zoom -
-        event.scale * (outFactor || 1) * Math.abs(velocity) * reduceFactor
-      );
-    }
+  // Go back to the previous screen
+  const goBack = () => {
+    navigation.goBack();
```

Comment:
I think you should make the navigation go back to the Home Screen directly so that we don't have : 
- go to the camera 
- go to the screen to post 
- click on retake button
- go to the camera
- go back 
- return to the camera instead of going to the homescreen

Reply:
Absolutely ! Impressive catch to be honest !